src/lib/utils.js

```diff
@@ -68,7 +68,7 @@ let utils = {
       if (item[property] === undefined) {
         throw new Error(`[groupBy]: Object has no key ${property}`);
       }
-      let key = item[property];
+      let key = item[property] && item[property].toLowerCase();
```

Comment:
I would make the code more explicit then.

```
let key = item[property];
if (key.toLowerCase) {
  let = key.toLowerCase();
}
```

Still not perfect (`key` can be an object with a `toLowerCase` field that is not a function...) but it expresses the intent of what you want to do more easily.